File: BSTIterator.js
class BSTIterator{
    constructor(root){
        this.q=[]
        this.allLeft(root)
    }
    hasNext(){
        return this.q.length>0
    }
    next(){
        let cur=this.q.pop()
        if(cur.right){
            this.allLeft(cur.right)
        }
        return cur.val
    }
    allLeft(root){
        while (root){
            this.q.push(root)
            root=root.left
        }
    }
}

File: BSTIterator.py
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None
class BSTIterator:
    def __init__(self, root):
        self.q=[]  
        self.allLeft(root)
    def hasNext(self):
        return len(self.q) > 0
    def next(self):
        cur = self.q.pop()
        if cur.right:
            self.allLeft(cur.right)
        return cur.val
    def allLeft(self, root):
        while root:
            self.q.append(root)
            root = root.left


